docs/USAGE-GUIDE.md

```diff
@@ -486,6 +486,30 @@ To use certificate transparency, the requester (usually the client) must call `s
 
 Call `s2n_connection_get_sct_list()` to retrieve the received certificate transparency information. The format of this data is the SignedCertificateTimestampList structure defined in section 3.3 of RFC 6962.
 
+## Session Resumption
+
+TLS handshake sessions are CPU-heavy due to the calculations involved in authenticating a certificate. These calculations can be skipped after the first connection by turning on session resumption. This mechanism stores state from the previous session and uses it to establish the next session, allowing the handshake to skip the costly authentication step while keeping the same cryptographic guarantees. Note that this session state is encrypted by the server, so a server will have to set up an external key in order to do session resumption.
+
+### Stateless Session Resumption
+
+In stateless session resumption the server creates a 'session ticket' from some state after a successful handshake. The server sends this ticket to a client, and the client can send that ticket back to the server during a new connection to skip the authentication step. This mechanism allows servers to avoid storing individual state for each client, and for that reason is the preferred method for resuming a session.
+
+#### A Note on TLS1.3 Session Resumption
+
+Session resumption works differently in versions TLS1.3 and higher. Session ticket messages are now sent immediately after the handshake in "post-handshake" messages, although more tickets can be sent and received anytime after the handshake has completed. Because of this clients need to call s2n_recv after negotiating to receive session ticket messages, as these could arrive anytime post-handshake. Additionally, multiple session tickets may be issued for the same connection.
+
+Servers should call `s2n_config_set_session_tickets_onoff()` to enable stateless session resumption. Additionally the server needs to set up an encryption key using `s2n_config_add_ticket_crypto_key()`.
+
+Clients should call `s2n_config_set_session_tickets_onoff()` to enable stateless session resumption and set a session ticket callback function using `s2n_config_set_session_ticket_cb()`. Then `s2n_connection_set_session()` should be called when attempting to resume a new connection.
+
+### Stateful Session Resumption
+
+Stateful session resumption, also known as session caching, is the same as the stateless mechanism, however in this case the server caches the session state per client and resumes a session based on the client's session ID. Note that session caching has not been implemented for > TLS1.2.
+
+Servers should call `s2n_config_set_session_cache_onoff()` to enable stateful session resumption, and set the three caching callback functions: `s2n_config_set_cache_store_callback()`, `s2n_config_set_cache_retrieve_callback()`, and `s2n_config_set_cache_delete_callback()`. Additionally, the server needs to set up an encryption key using `s2n_config_add_ticket_crypto_key()`.
```

Comment:
To confirm: you DON'T need to also call s2n_config_set_session_tickets_onoff?

Reply:
No... you don't additionally need to call that function and turn session tickets off.

Reply:
No, I meant do you need to turn tickets on before you turn caching on, or does turning caching on both enable tickets and turn on the session cache. I remember there was some confusion there, don't remember the answer.

Reply:
You turn these two features on individually. Why would turning on caching also turn on tickets?